aten/src/ATen/miopen/Descriptors.cpp

```diff
@@ -109,9 +109,25 @@ void FilterDescriptor::set(const at::Tensor &t, int64_t pad) {
   for (int i = dim; i < pad; ++i) {
     size[i] = (int) 1;
   }
-  for (int i = dim - 1; i >=0; --i) {
-    stride[i] = (i == dim - 1) ? 1 : stride[i+1] * size[i+1];
+  if( memory_format != at::MemoryFormat::ChannelsLast ) {
+      for (int i = pad; i >= dim; --i ) {
+          stride[i] = 1;
+      }
+      for (int i = dim - 1; i >=0; --i) {
+        stride[i] = (i == dim - 1) ? 1 : stride[i+1] * size[i+1];
+      }
   }
+  else {
+      // ChannelsLast
+       for (int i = pad; i >= dim; --i ) {
+          stride[i] = 1;
+      }
+      for (int i = dim-1 ; i >=0; --i ) {
+          // Pass-through
+          stride[i] = t.stride(i);
+      }
+  }
```

Comment:
I'm confused by this. Why don't you just read out the strides from the tensor, instead of recomputing them here? (Does miopen require some specific canonical form of strides?)

Reply:
For the Pad dimensions, it needs to be set to 1. So, this addition is needed. 
In the ChannelsLast code, the new code is essentially doing what you propose, that is, reading out strides from the tensor. 
In the NCHW (and other code), on line 117, I only kept it to recompute because the original code was recomputing. I don't think we need to recompute. If you are also on same opinion, I can change both code to just pass-through

Reply:
OK I see. You don't have to fix the original code in this PR but I would like to see it fixed (Unless there is a reason for recompute, in which case it should be commented why)

Reply:
I decided to fix the original code; Tested locally and test_nn passes. Lets see if CI tests pass. If it does, we are good.